plutus-core-spec/plutus-core-specification.tex

```diff
@@ -1371,6 +1378,141 @@ \subsubsection{Comparison of the typed CK machine and the untyped CEK machine}
 closure $C$ with $\discharge{C} = \erase{V}$.
 \end{itemize}
 
+\section{Serialisation}
+
+We use the \texttt{flat} format to serialise plutus terms. The \texttt{flat} format
+encodes sum types as tagged unions and products by concatenating their contents. We
+proceed by defining the structure and the data types of untyped plutus core and how
+they get serialised.
+
+\subsection{Variable length data}
+
+\texttt{Lists} are encoded by prefixing (using the most significant bit) the element
+stored in the list with one of the two list constructor tags. If we want to follow the
+current element with another one we use the bit `1` for the \texttt{Cons} constructor,
+or `0` for the \texttt{Nil} constructor if this is the last element of the list.
+
+\noindent We encode variable length data as a list of chunks, 7 bits each, with the least
+significant chunk first and the most significant bit first in the chunk.
```

Comment:
I have checked the serialisation output and everything seems fine. I wrote documentation of the steps I took to verify, so we can continue the discussion using that file instead of this one.

Reply:
After spending a bit of time unwinding the serialised format it looks like it uses one byte for constants that can be encoded in 7bits (do you think we should optimise this?). Maybe things will make more sense after you will read through the example serialisation of `(program 11.22.33 (con integer 11))`.